specification/monitor/Azure.Monitor.Query.Logs/models.tsp

```diff
@@ -0,0 +1,674 @@
+namespace MonitorQueryLogs;
+
+/** The data type of a column. */
+union ColumnDataType {
+  string,
+
+  /** Boolean data type */
+  bool: "bool",
+
+  /** DateTime data type */
+  datetime: "datetime",
+
+  /** Dynamic data type */
+  dynamic: "dynamic",
+
+  /** Integer data type */
+  int: "int",
+
+  /** Long integer data type */
+  long: "long",
+
+  /** Real/floating point data type */
+  real: "real",
+
+  /** String data type */
+  string: "string",
+
+  /** GUID data type */
+  guid: "guid",
+
+  /** Decimal data type */
+  decimal: "decimal",
+
+  /** Timespan data type */
+  timespan: "timespan",
+}
+
+/** Contains the tables, columns & rows resulting from a query. */
+model QueryResults {
+  /** The results of the query in tabular format. */
+  tables: Table[];
+
+  /** Statistics represented in JSON format. */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "This is an arbitrary object"
+  statistics?: Record<unknown>;
+
+  /** Visualization data in JSON format. */
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type" "This is an arbitrary object"
+  render?: Record<unknown>;
+
+  /** The code and message for an error. */
+  error?: ErrorInfo;
+}
+
+/** Contains the columns and rows for one table in a query response. */
+model Table {
+  /** The name of the table. */
+  name: string;
+
+  /** The list of columns in this table. */
+  columns: Column[];
+
+  /** The resulting rows from this query. */
+  rows: Record<unknown>[][];
```

Comment:
For Java, we need to use `unknown[][]` to generate the same type as we did with Swagger. However, we can do this through `@alternateType` customization if this works for all other languages.